pallets/collator-selection/src/lib.rs

```diff
@@ -555,6 +578,126 @@ pub mod pallet {
 			Self::deposit_event(Event::InvulnerableRemoved { account_id: who });
 			Ok(())
 		}
+
+		/// Todo
+		///
+		/// Todo
+		#[pallet::call_index(7)]
+		#[pallet::weight(T::WeightInfo::increase_bond(T::MaxCandidates::get()))]
```

Comment:
If we keep the candidates in a vec, perhaps we should do weight refund here using the `DispatchWeightWithPostInfo`, as the worst case estimate may be overestimating and fees will be too high. See more [here](https://substrate.stackexchange.com/questions/9246/what-are-the-consequences-of-overestimating-weight) and maybe some helpful pointers [here](https://hackmd.io/@gpestana/S1g4PySDj)

Reply:
Done.